src/client/pythonEnvironments/discovery/locators/services/condaService.ts

```diff
@@ -422,8 +420,12 @@ export class CondaService implements ICondaService {
         if (await inDiscoveryExperiment(this.experimentService)) {
             return this.pyenvs.getWinRegInterpreters(undefined);
         }
-        if (this.registryLookupForConda) {
-            return this.registryLookupForConda.getInterpreters();
+
+        const registryLookupForConda: IInterpreterLocatorService | undefined = this.serviceContainer.get<
```

Comment:
This will throw error when in experiment and WINDOWS_REGISTRY_SERVICE is not found, we should put this in the else part.

Reply:
When in experiment we will return, it won't reach this line.

Reply:
Oh, yes. Although do we need the `undefined` in the type?
```suggestion
        const registryLookupForConda: IInterpreterLocatorService = this.serviceContainer.get<
```